simpleflow/swf/mapper/core.py

```diff
@@ -0,0 +1,612 @@
+# Copyright (c) 2013, Theo Crevon
+# Copyright (c) 2013, Greg Leclercq
+#
+# See the file LICENSE for copying permission.
+from __future__ import annotations
+
+from datetime import datetime
+import os
+from typing import Any
+
+import boto3
+from botocore.exceptions import NoCredentialsError
+
+# NB: import logger directly from simpleflow so we benefit from the logging
+# config hosted in simpleflow. This wouldn't be the case with a standard
+# "logging.getLogger(__name__)" which would write logs under the "swf" namespace
+from simpleflow import logger
+from simpleflow.utils import remove_none, retry
+
+from . import settings
+
+SETTINGS = settings.get()
+RETRIES = int(os.environ.get("SWF_CONNECTION_RETRIES", "5"))
+DEFAULT_AWS_REGION = "us-east-1"
+
+
+class ConnectedSWFObject:
+    """Authenticated object interface"""
+
+    __slots__ = ["region", "connection", "boto3_client"]
+
+    region: str
+    boto3_client: boto3.client
+
+    @retry.with_delay(
+        nb_times=RETRIES,
+        delay=retry.exponential,
+        on_exceptions=(TypeError, NoCredentialsError),
+    )
+    def __init__(self, *args, **kwargs):
+        self.region = SETTINGS.get("region") or kwargs.get("region") or DEFAULT_AWS_REGION
+        # Use settings-provided keys if available, otherwise pass empty
+        # dictionary to boto SWF client, which will use its default credentials
+        # chain provider.
+        cred_keys = ["aws_access_key_id", "aws_secret_access_key"]
+        creds_ = {k: SETTINGS[k] for k in cred_keys if SETTINGS.get(k, None)}
+
+        self.boto3_client = kwargs.pop("boto3_client", None)
+        if not self.boto3_client:
+            session = boto3.session.Session(region_name=self.region)
+            # raises EndpointConnectionError if region is wrong
+            self.boto3_client = session.client("swf", **creds_)
+
+        logger.debug(f"initiated connection to region={self.region}")
+
+    # Proxy for https://boto.cloudhackers.com/en/latest/ref/swf.html#boto.swf.layer1.Layer1.list_open_workflow_executions
+    # written with boto3.
```

Comment:
Do we still need these comments? I understand their usefulness (and the functions themselves) in previous iterations of the PR, but now? Mind you, no opinion either way, just curious

Reply:
Yay they're not useful at all, with maybe an exception on "list_open_workflow_executions" and "list_closed_workflow_executions" where I reaaally tried to mimic the boto2 behaviour.